### PR TITLE
Adding qt project generation fix #7784

### LINKED ISSUE #7784: [qt5cpp] Generate project include file (*.pri)
##### Description

This would be really nice to have the project include file generated! Example for the Pet store example:

    QT += network
    
    HEADERS += \
    	$${PWD}/SWGApiResponse.h \
    	$${PWD}/SWGCategory.h \
    	$${PWD}/SWGHelpers.h \
    	$${PWD}/SWGHttpRequest.h \
    	$${PWD}/SWGModelFactory.h \
    	$${PWD}/SWGObject.h \
    	$${PWD}/SWGOrder.h \
    	$${PWD}/SWGPet.h \
    	$${PWD}/SWGPetApi.h \
    	$${PWD}/SWGStoreApi.h \
    	$${PWD}/SWGTag.h \
    	$${PWD}/SWGUser.h \
    	$${PWD}/SWGUserApi.h
    
    SOURCES += \
    	$${PWD}/SWGApiResponse.cpp \
    	$${PWD}/SWGCategory.cpp \
    	$${PWD}/SWGHelpers.cpp \
    	$${PWD}/SWGHttpRequest.cpp \
    	$${PWD}/SWGOrder.cpp \
    	$${PWD}/SWGPet.cpp \
    	$${PWD}/SWGPetApi.cpp \
    	$${PWD}/SWGStoreApi.cpp \
    	$${PWD}/SWGTag.cpp \
    	$${PWD}/SWGUser.cpp \
    	$${PWD}/SWGUserApi.cpp

##### Swagger-codegen version

2.3.1

##### Swagger declaration file content or url

https://gist.github.com/MartinDelille/c358209832ed023c1cd7ddb308db0099

##### Command line used for generation

    $ swagger-codegen generate -i api.yml -l qt5cpp

##### Steps to reproduce

None it is a feature request

##### Related issues/PRs

Not that I know.

##### Suggest a fix/enhancement

I'll be happy to make a PR if you provide me some little support!

### PR DESCRIPTION
**Pull request by MartinDelille**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

I would like to fix #7784 by adding `client.pri` generation. This is a work in progress but I open this PR to address some problem I have with its implementation.

For the Pet Store example, the resulting file should look like this:

```
QT += network

HEADERS += \
	$${PWD}/SWGApiResponse.h \
	$${PWD}/SWGCategory.h \
	$${PWD}/SWGHelpers.h \
	$${PWD}/SWGHttpRequest.h \
	$${PWD}/SWGModelFactory.h \
	$${PWD}/SWGObject.h \
	$${PWD}/SWGOrder.h \
	$${PWD}/SWGPet.h \
	$${PWD}/SWGPetApi.h \
	$${PWD}/SWGStoreApi.h \
	$${PWD}/SWGTag.h \
	$${PWD}/SWGUser.h \
	$${PWD}/SWGUserApi.h

SOURCES += \
	$${PWD}/SWGApiResponse.cpp \
	$${PWD}/SWGCategory.cpp \
	$${PWD}/SWGHelpers.cpp \
	$${PWD}/SWGHttpRequest.cpp \
	$${PWD}/SWGOrder.cpp \
	$${PWD}/SWGPet.cpp \
	$${PWD}/SWGPetApi.cpp \
	$${PWD}/SWGStoreApi.cpp \
	$${PWD}/SWGTag.cpp \
	$${PWD}/SWGUser.cpp \
	$${PWD}/SWGUserApi.cpp
```

Here are the tasks to do:

- [x] Add `optionalProjectFile` switch to the command line
- [ ] Generate `client.pri`
- [ ] Add test

Ping @ravinikam @stkrwork  @fvarose

### CODE REVIEW

**Comment by MartinDelille** [comment]
As you can see, the tests are failing:
https://travis-ci.org/swagger-api/swagger-codegen/builds/350829855#L2556

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Qt5CPPGenerator.java [hunk 1]
```diff
@@ -114,6 +117,9 @@ public Qt5CPPGenerator() {
         supportingFiles.add(new SupportingFile("HttpRequest.cpp.mustache", sourceFolder, PREFIX + "HttpRequest.cpp"));
         supportingFiles.add(new SupportingFile("modelFactory.mustache", sourceFolder, PREFIX + "ModelFactory.h"));
         supportingFiles.add(new SupportingFile("object.mustache", sourceFolder, PREFIX + "Object.h"));
+        if (optionalProjectFileFlag) {
+            supportingFiles.add(new SupportingFile("Project.mustache", "", "client.pri"));
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Qt5CPPGenerator.java]: you are missing the mustache file

Comment (stkrwork) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Qt5CPPGenerator.java]: also check the processOpts function

**Comment by MartinDelille** [comment]
travis build failed but it doesn't seems to be related to my modification, am I right?

Now I just need to fill `Project.mustache` file but since I'm new to mustache, someone could give me a clue how to list the headers and sources files generated?

**Comment by stkrwork** [comment]
`
QT += network

HEADERS += \
{{#operations}}
	$${PWD}/{{classname}.h \
{{/operations}}
{{#models}}
{{#model}}
	$${PWD}/{{classname}.h \
{{/model}}
{{/models}}

SOURCES += \
{{#operations}}
	$${PWD}/{{classname}.cpp \
{{/operations}}
{{#models}}
{{#model}}
	$${PWD}/{{classname}.cpp \
{{/model}}
{{/models}}
`

I think, but i am not sure

**Comment by MartinDelille** [comment]
Ok I implemented the mustache part. Let me know if it looks good to you!

Ping @ravinikam @stkrwork @fvarose

File: modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# APIs
+{{#apiInfo}}
+{{#apis}}
+{{#operations}}
+    $${PWD}/{{classname}}.cpp \
+{{/operations}}
+{{/apis}}
+{{/apiInfo}}
+# Others
+    $${PWD}/SWGHelpers.cpp \
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache]: $${PWD}/{{prefix}}Helpers.cpp \

File: modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    $${PWD}/SWGHttpRequest.cpp
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache]: same as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# APIs
+{{#apiInfo}}
+{{#apis}}
+{{#operations}}
+    $${PWD}/{{classname}}.h \
+{{/operations}}
+{{/apis}}
+{{/apiInfo}}
+# Others
+    $${PWD}/SWGHelpers.h \
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache]: see below

File: modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache (hunk 2 above)
Comment (MartinDelille) [reply modules/swagger-codegen/src/main/resources/qt5cpp/Project.mustache]: Done

**Comment by wing328** [comment]
@MartinDelille thanks for the PR, which has been merged into master.

cc @etherealjoy as well

**Pull request merged**